agent/src/main/java/io/pyroscope/javaagent/impl/ContinuousProfilingScheduler.java

```diff
@@ -67,6 +67,15 @@ public void start(Profiler profiler) {
 
     }
 
+    /**
+     * Stops the profiling scheduler which stops the AsyncProfiler.
+     * @param profiler
+     */
+    @Override
+    public void stop(Profiler profiler) {
+        profiler.stop();
```

Comment:
does this stop the `this.job` task somehow ?
What happens if this stop method is executed concurrently with `dumpProfile` runnable?  Is it possible? Are they synchronized in any way? should they?

Reply:
I have added the logic to stop `this.job` task as well.  Let me look into the second part of running stop concurrently with dumpProfile runnable.

Reply:
I am not sure if the dumProfile and stop should be synchronized. @korniltsev any insights?

Reply:
I think it should. 

T(dump) calls profiler.stop
T(stopper) calls scheduler.stop 
T(dump) calls profiler.start

And now it profiles forever

Reply:
Ok let me make that change,

Reply:
I have made the change @korniltsev